pkg/automationconfig/automation_config_builder.go

```diff
@@ -0,0 +1,89 @@
+package automationconfig
+
+import (
+	"fmt"
+	"github.com/spf13/cast"
+)
+
+type Topology string
+
+const (
+	ReplicaSetTopology Topology = "ReplicaSet"
+)
+
+type Builder struct {
+	processes      []Process
+	replicaSets    []ReplicaSet
+	version        string
+	auth           Auth
+	members        int
+	domain         string
+	name           string
+	topology       Topology
+	mongodbVersion string
+}
+
+func NewBuilder() *Builder {
+	return &Builder{
+		processes:   []Process{},
+		replicaSets: []ReplicaSet{},
+	}
+}
+
+func (b *Builder) SetTopology(topology Topology) *Builder {
+	b.topology = topology
+	return b
+}
+
+func (b *Builder) SetMembers(members int) *Builder {
+	b.members = members
+	return b
+}
+
+func (b *Builder) SetDomain(domain string) *Builder {
+	b.domain = domain
+	return b
+}
+
+func (b *Builder) SetName(name string) *Builder {
+	b.name = name
+	return b
+}
+
+func (b *Builder) SetMongoDBVersion(version string) *Builder {
+	b.mongodbVersion = version
+	return b
+}
+
+func (b *Builder) SetAutomationConfigVersion(version string) *Builder {
+	b.version = version
+	return b
+}
+
+func (b *Builder) Build() AutomationConfig {
+	hostnames := make([]string, b.members)
+	for i := 0; i < b.members; i++ {
+		hostnames[i] = fmt.Sprintf("%s-%d.%s", b.name, i, b.domain)
```

Comment:
should we have a function to create hostname: we're already duplicating that code

![](https://media.giphy.com/media/NEJCPYzr6xmAU/giphy.gif)

Reply:
yes good point, will add a small helper function